File: midterm lab 3/java/MP08.java
import java.io.*;
import java.util.*;

// MP08 - Filter records using a keyword
public class MP08 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        try {
            // Ask user for file path
            System.out.print("Enter CSV file path: ");
            String filePath = scanner.nextLine();

            // Ask for keyword
            System.out.print("Enter keyword to filter: ");
            String keyword = scanner.nextLine().toLowerCase();

            BufferedReader br = new BufferedReader(new FileReader(filePath));
            String line;
            int count = 0;

            System.out.println("\nFiltered Records:");

            // Read CSV line by line
            while ((line = br.readLine()) != null) {
                // Check if line contains keyword
                if (line.toLowerCase().contains(keyword)) {
                    System.out.println(line);
                    count++;
                }
            }

            System.out.println("\nTotal matched records: " + count);
            br.close();

        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}

File: midterm lab 3/javascript/MP08.js
// MP08 - Filter records using a keyword

const fs = require("fs");
const readline = require("readline");

// Create interface for user input
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Ask for file path
rl.question("Enter CSV file path: ", (filePath) => {
    rl.question("Enter keyword to filter: ", (keyword) => {
        try {
            const data = fs.readFileSync(filePath, "utf-8");

            const lines = data.split("\n");
            let count = 0;

            console.log("\nFiltered Records:");

            // Loop through each row
            lines.forEach(line => {
                if (line.toLowerCase().includes(keyword.toLowerCase())) {
                    console.log(line);
                    count++;
                }
            });

            console.log("\nTotal matched records:", count);

        } catch (err) {
            console.log("Error:", err.message);
        }

        rl.close();
    });
});